torch/onnx/symbolic.py

```diff
@@ -961,8 +961,18 @@ def retrieve_state(x, start, end):
                                                  **extra_kwargs)
 
             if bidirectional:
-                prev_output = g.op('Reshape', prev_output, g.op('Constant', value_t=torch.LongTensor([0, 1, 0, -1])))
-            prev_output = g.op('Squeeze', prev_output, axes_i=[1])
+                # The ONNX RNN/GRU/LSTM produce an output of dimensions
+                #   seq_len, num_directions, batch, hidden_size
+                # We have to convert to match pytorch's expected
+                #   seq_len, batch, hidden_size * num_directions
+                # by first moving num_directions to the end with
+                # Transpose, and then combining it with hidden_size
+                # with Reshape.
+                prev_output = g.op('Transpose', prev_output, perm_i=[0, 2, 3, 1])
+                prev_output = g.op('Reshape', prev_output, g.op('Constant', value_t=torch.LongTensor([0, 0, -1, 1])))
+                prev_output = g.op('Squeeze', prev_output, axes_i=[3])
```

Comment:
If you're going to do the `Squeeze` in the unidirectional case only rather than both cases, this can just be `prev_output = g.op('Reshape', prev_output, g.op('Constant', value_t=torch.LongTensor([0, 0, -1])))`